server/build.gradle

```diff
@@ -411,14 +411,21 @@ tasks.register("downloadSnapshot", Copy) {
     def groupId = "org.opensearch"
     def artifactId = "opensearch"
 
-    repositories {
-        maven {
-            url mavenSnapshotRepoUrl
-        }
-    }
+    def repos = project.getRepositories();
```

Comment:
@reta Why did we need to make these changes?  It looks like it caused issues in the download workflow [1] for other PRs.


```
./gradlew :server:japicmp
...
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':server:downloadSnapshot'.
> Could not resolve all task dependencies for configuration ':server:snapshotArtifact'.
   > Could not find org.apache.lucene:lucene-core:9.11.0-snapshot-fb[97](https://github.com/opensearch-project/OpenSearch/actions/runs/8693672822/job/23846998626?pr=12812#step:4:100)840.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/apache/lucene/lucene-core/9.11.0-snapshot-fb97840/lucene-core-9.11.0-snapshot-fb97840.pom
       - https://aws.oss.sonatype.org/content/repositories/snapshots/org/apache/lucene/lucene-core/9.11.0-snapshot-fb97840/lucene-core-9.11.0-snapshot-fb97840.pom
```

- [1] https://github.com/opensearch-project/OpenSearch/actions/runs/8693672822/job/23846998626?pr=12812

Reply:
@peternied the change was supposed to fix exactly that: the issue I run into was that that snapshot tasks tries to download opensearch deps from the `https://aws.oss.sonatype.org/` whereas it should not since it has to use another repo. The change adds the `https://aws.oss.sonatype.org/` as exclusive for `org.opensearch` so it seemed to help but not entirely, I will take a look shortly.

Reply:
Thanks.  I'm curious if the source for the artifacts changed.  We have a myriad of changes [1] with hard to predict downstream effects, but this seemed like a new source or at least a source we haven't seen create friction before.

- [1] https://github.com/opensearch-project/OpenSearch/issues/2350

Reply:
> Thanks. I'm curious if the source for the artifacts changed

The Apache Lucene snapshots are coming from different place (`RepositoriesSetupPlugin`): https://ci.opensearch.org/ci/dbc/snapshots/lucene/